### PR TITLE
Add test cases of search iterator

### PR DESCRIPTION
**Pull request by NicoYuan1986**
issue: #25028 #25022 #25003

### CODE REVIEW

File: tests/python_client/testcases/test_query.py [hunk 1]
```diff
@@ -1618,6 +1618,100 @@ def test_query_using_all_types_of_default_value(self):
...
+        """
+        # 1. initialize with data
+        limit = 100
+        collection_w = self.init_collection_general(prefix, True, is_index=False)[0]
+        collection_w.create_index(ct.default_float_vec_field_name, {"metric_type": "L2"})
+        collection_w.load()
+        # 2. search iterator
+        expr = "int64 >= 0"
+        query_iterator = collection_w.query_iterator(expr, limit=limit)[0]
+        # 3. check the result
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:1637]: Could we write check function in the check module of our framework?

File: tests/python_client/testcases/test_query.py [hunk 2]
```diff
@@ -1618,6 +1618,100 @@ def test_query_using_all_types_of_default_value(self):
         assert res[ct.default_bool_field_name] is False
         assert res[ct.default_string_field_name] == "abc"
 
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_query_iterator_normal(self):
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:1622]: Here is all valid cases, is the invalid cases (like params check test case) in your next pr?

File: tests/python_client/testcases/test_search.py [hunk 3]
```diff
@@ -8359,3 +8359,270 @@ def test_search_json_expression_default(self, nq, dim, auto_id, is_flush, enable
...
+        while True:
+            res = search_iterator.next()
+            if len(res[0]) == 0:
+                log.info("search iteration finished, close")
+                search_iterator.close()
+                break
+            for i in range(len(res[0])):
+                # log.info(res[0][i])
+                pk_list.append(res[0][i].id)
+            page_idx += 1
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:8627]: what is the function of the "page_idx"?

File: tests/python_client/check/func_check.py [hunk 4]
```diff
@@ -282,6 +290,46 @@ def check_search_results(search_res, func_name, check_items):
...
+                2. check the limit(topK) and ids
+                3. check the distance
+        expected: check the search is ok
+        """
+        log.info("search_iterator_results_check: checking the searching results")
+        if func_name != 'search_iterator':
+            log.warning("The function name is {} rather than {}".format(func_name, "search_iterator"))
+        search_iterator = search_res
+        pk_list = []
+        while True:
```
Comment (yanliang567) [inline tests/python_client/check/func_check.py:307]: when this loop will exit?

**Review by yanliang567 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**